test/controllers/preflight.test.js

```diff
@@ -16,10 +16,20 @@ import { use, expect } from 'chai';
 import chaiAsPromised from 'chai-as-promised';
 import sinonChai from 'sinon-chai';
 import sinon from 'sinon';
+import { Site as SiteModel } from '@adobe/spacecat-shared-data-access';
+import ImsPromiseClient from '@adobe/spacecat-shared-ims-client/src/clients/ims-promise-client.js';
```

Comment:
@anagarwa , You're not using `ImsPromiseClient` in `src/controllers/preflight.js`. Since we're writing unit tests, you should mock `getCSPromiseToken` from [support/utils.js](https://github.com/adobe/spacecat-api-service/pull/978/files#diff-8ee4f6d0c5f6cec11896da192935049545946e2174fe4b103102b8d0aa9f1ce9R20); otherwise, the tests become coupled to the internal implementation of `getCSPromiseToken` - specifically, `ImsPromiseClient`. If that implementation ever changes, our tests will break.